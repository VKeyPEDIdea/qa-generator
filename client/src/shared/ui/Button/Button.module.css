.btn {
    color: var(--color-brown);
    font-size: 14px;
    background-color: transparent;
    border: 1px solid var(--color-card-border);
    border-radius: 8px;
    text-transform: uppercase;
    padding: 8px 16px;
    cursor: pointer;
    font-weight: 600;
    font-family: var(--primary-font);
    transition: var(--st-trans);

  }
.btn:hover {
    background-color: var(--color-paper-light);
    border: 1px solid var(--color-brown);
}