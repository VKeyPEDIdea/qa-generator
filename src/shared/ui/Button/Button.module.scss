@import 'app/styles/vars';

.btn {
    color: $color-brown;
    font-size: 14px;
    background-color: transparent;
    border: 1px solid $color-card-border;
    border-radius: 8px;
    text-transform: uppercase;
    padding: 8px 16px;
    cursor: pointer;
    font-weight: 600;
    font-family: $primary-font;
    transition: $st-trans;

    &:hover {
        background-color: $color-paper-light;
        border: 1px solid $color-brown;
    }
}