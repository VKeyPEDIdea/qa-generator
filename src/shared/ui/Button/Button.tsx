interface ButtonProps {
    title: string;
    onClick: () => void;
}

const Button = ({
    title,
    onClick,
}: ButtonProps) => {
    return (
        <button onClick={onClick}>
            <span>{title}</span>
        </button>
    );
};

export default Button;