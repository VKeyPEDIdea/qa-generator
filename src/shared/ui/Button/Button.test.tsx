import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import Button from './Button';

describe('Button component', () => {
  it('should render with passed text', () => {
    const title = 'Нажми меня!';
    const { getByText } = render(<Button title={title} />);
    const buttonElement = getByText(title);
    expect(buttonElement).toBeInTheDocument();
  });

  it('should call "onClick" function when click', () => {
    const onClickMock = vi.fn();
    const { getByText } = render(
      <Button title="Тестовая кнопка" onClick={onClickMock} />,
    );
    const buttonElement = getByText('Тестовая кнопка');

    fireEvent.click(buttonElement);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('should render correctly', () => {
    const { container } = render(<Button title="Нажми меня!" />);
    expect(container).toMatchSnapshot();
  });
});
