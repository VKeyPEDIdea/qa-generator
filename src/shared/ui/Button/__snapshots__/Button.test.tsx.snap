// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should render correctly 1`] = `
<div>
  <button
    class="_btn_0bbf54"
  >
    <span>
      Нажми меня!
    </span>
  </button>
</div>
`;
